driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -707,9 +708,9 @@ public void onSuccess(ResultSet result) {
 
             @Override
             public void onFailure(Throwable t) {
-              // downgrade to system.peers if we get an invalid query as this indicates
-              // the peers_v2 table does not exist.
-              if (t instanceof InvalidQueryException) {
+              // downgrade to system.peers if we get an invalid query or server error as this
+              // indicates the peers_v2 table does not exist.
+              if (t instanceof InvalidQueryException || t instanceof ServerError) {
```

Comment:
Didn't we decide to also match the error message in the case of `ServerError`?

Reply:
My mistake, that's right.  I'll add that.